src/vaadin-combo-box-mixin.html

```diff
@@ -352,7 +357,7 @@
       } else if (path.indexOf(this.inputElement) !== -1) {
         if (path.indexOf(this._toggleElement) > -1 && this.opened) {
           this.close();
-        } else {
+        } else if (path.indexOf(this._toggleElement) > -1 || !this.autoOpenDisabled) {
           this.open();
```

Comment:
not sure but IMOP, it should be refactored like 
`if ((path.indexOf(this._toggleElement) > -1) {
if (this.opened) {
this.close();
}
else if (!this.autoOpenDisabled) {
this.open();
}
`

Reply:
Hey I tried to edit your comment so the code was clearer, hope you don't mind

Reply:
In the original code, the outer `if` is checking for the presence of the `inputElement` and the inner ones are checking for the `toggleButton`, I don't think your refactor accounts for all that logic. There might be a way to make this cleaner but my approach was to add on in a way that changed as little as possible.

Reply:
Thank you for editing my comment. :) yes. :) I understand your mean.